azure-devops/run_build.yml

```diff
@@ -14,12 +14,22 @@ jobs:
   steps:
     - checkout: self
       submodules: recursive
+    - task: run-cmake@0
```

Comment:
@lukka Do you know why this is causing the run-cmake bits to select clang instead of msvc?

```
[command]"C:\Program Files\CMake\bin\cmake.exe" -G Ninja "-DCMAKE_MAKE_PROGRAM=C:\Program Files\CMake\bin\ninja.exe" -DCMAKE_BUILD_TYPE=Debug D:\agent\_work\1\s\tools
-- The CXX compiler identification is Clang 8.0.1 with GNU-like command-line
CMake Error at C:/Program Files/CMake/share/cmake-3.15/Modules/Platform/Windows-Clang.cmake:81 (enable_language):
  No CMAKE_RC_COMPILER could be found.

  Tell CMake where to find the compiler by setting either the environment
  variable "RC" or the CMake cache entry CMAKE_RC_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.15/Modules/Platform/Windows-Clang.cmake:121 (__windows_compiler_clang_gnu)
  C:/Program Files/CMake/share/cmake-3.15/Modules/Platform/Windows-Clang-CXX.cmake:3 (__windows_compiler_clang)
  C:/Program Files/CMake/share/cmake-3.15/Modules/CMakeCXXInformation.cmake:48 (include)
  CMakeLists.txt:5 (project)


-- Configuring incomplete, errors occurred!
See also "D:/agent/_work/1/a/tools/CMakeFiles/CMakeOutput.log".
##[error]Error: CMake failed with error: '1'.
##[error]CMake failed with error: 'Error: CMake failed with error: '1'.'.
##[section]Finishing: Build STL Support Tools
```

Reply:
Looks like the run-vcpkg task has some sidechannel here?

Reply:
@BillyONeal 
yes, `run-vcpkg` sets where vcpkg has been installed and the default triplet. Then `run-cmake` uses `vcpkg env` command to setup the proper environment. E.g.  the most recent build that correctly selected MSVC you can see the `vcpkg env` is run by `run-cmake` in the [build log](https://dev.azure.com/vclibs/STL/_build/results?buildId=634&view=logs&j=3c8766a0-af1e-5776-073c-45404a320db8&t=ae72f6b1-492c-55c2-25e3-2cd45a58a842&l=9). 
It is documented in the [overview.md](https://marketplace.visualstudio.com/items?itemName=lucappa.cmake-ninja-vcpkg-tasks) file. I think I should organize docs better.

Reply:
I admit to just trying to copy paste your contribution and change a few places randomly here, so the docs might be fine.

Reply:
The thing that's odd about that is it says it's using that `CMakeSettings.txt` which has no clang configurations. (That is, it seems to be totally ignoring CMakeSettings.txt and just using what vcpkg thinks)